### PR TITLE
UX tweaks

### PR DESCRIPTION
**Pull request by corbt**
Bunch of UX tweaks, mostly to the Evaluate page.

Screenshot 2023-11-30 at 11 19 28 PM

### CODE REVIEW

File: app/src/components/datasets/DatasetContentTabs/Evaluation/FormattedMessage.tsx [hunk 1]
```diff
@@ -1,66 +1,79 @@
...
-            style: { wordBreak: "break-all", whiteSpace: "pre-wrap" },
-          }}
-          wrapLines
-        >
-          {JSON.stringify(JSON.parse(args), null, 4)}
-        </SyntaxHighlighter>
-      </VStack>
-    );
-  }
+    formattedJson = JSON.stringify(JSON.parse(props.json), null, 2);
```
Comment (arcticfly) [inline app/src/components/datasets/DatasetContentTabs/Evaluation/FormattedMessage.tsx:54]: What happens if `json` isn't actually parseable to json? Do we render nothing?

File: app/src/server/api/routers/datasetEntries.router.ts [hunk 2]
```diff
@@ -647,21 +647,21 @@ export const datasetEntriesRouter = createTRPCRouter({
...
-              `CASE
+          .orderBy(
+            () =>
+              sql`CASE
                 WHEN "averageScoreForEval"."datasetEntryId" IS NULL THEN 1
                 ELSE 0
               END`,
-            ),
           )
-          .orderBy(`averageScoreForEval.score`, sortOrder.order) as unknown as typeof baseQuery;
```
Comment (arcticfly) [inline app/src/server/api/routers/datasetEntries.router.ts:642]: You're right, we didn't need that hacky typing here.

**Review by arcticfly — APPROVED** [review]
Makes sense 👍

File: app/src/components/datasets/DatasetContentTabs/Evaluation/FormattedMessage.tsx (hunk 1 above)
Comment (corbt) [reply app/src/components/datasets/DatasetContentTabs/Evaluation/FormattedMessage.tsx:54]: If it isn't valid JSON it just falls back to to using the unformatted string (but still tries to highlight it as code).

**Pull request merged**